node_modules/col-core/lib/api.js

```diff
@@ -51,9 +52,61 @@ var init = module.exports.init = function(callback) {
  */
 var initializeServer = function(callback) {
   // Initialize the Express server
-  module.exports.appServer = Server.setUpServer();
-  // Initialize the REST router wrapper
-  module.exports.appRouter = Server.setupRouter(module.exports.appServer);
+  var appServer = module.exports.appServer = Server.setUpServer();
+
+  // A router for all routes on /lti
+  var ltiRouter = module.exports.ltiRouter = express.Router();
+  appServer.use('/lti', ltiRouter);
+
+  // A router for all routes on /api/courses/:api_domain_course_id
+  var apiRouter = module.exports.apiRouter = express.Router();
+
+  // This middleware will check if the user is authorized to interact with the given course
+  apiRouter.use(function(req, res, next) {
+    // The base URL in this middleware will be /api/courses/ucberkeley.canvas.com~21312
+    // Unfortunately, express doesn't parse the api_domain and course in middleware, so
+    // we need to do some manual string mangling
+    var apiDomainCourseId = req.baseUrl.substring(13);
+    var apiDomain = apiDomainCourseId.split('~')[0];
+    var courseId = apiDomainCourseId.split('~')[1];
+
+    // Get the user id from the client's cookies
```

Comment:
Should we explain that there will be a separate cookie per course+canvas_instance the user is in?